### PR TITLE
Use Github actions instead Semaphore CI

### PR DESCRIPTION
**Pull request by EvgeniyEsaulkov**
### Summary

This PR replaces Semaphore CI with Github actions

Also:
- automatic deploy to Heroku will be done in follow-up PR

### Review notes

While reviewing pull-request (especially when it's your pull-request),
please make sure that:

- you understand what problem is solved by PR and how is it solved
- new tests are in place, no redundant tests
- DB schema changes reflect new migrations
- newly introduced DB fields have indexes and constraints
- there are no missed files (migrations, view templates)
- required ENV variables added and described in `.env.example` and added to Heroku
- associated Heroku review app works correctly with introduced changes

### Deploy notes

Notes regarding deployment the contained body of work.
These should note any db migrations, ENV variables, services, scripts, etc.

### References
* Resolves #### Issue
* [GitHub Pull Request ####](https://github.com/fs/rails-base-graphql-api/pull/####)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        with:
+          bundler-cache: true
+
+      - name: Run quality checks
+        run: |
+          bin/quality
+
+      - name: Setup db
+        run: |
+          sudo apt-get -yqq install libpq-dev
```
Comment (DmitryBarskov) [inline .github/workflows/ci.yml:44]: Why do you need this line? I have projects working without it

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          bin/rails db:create db:schema:load
```
Comment (DmitryBarskov) [inline .github/workflows/ci.yml:45]: maybe `bin/setup` is a better alternative here?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        uses: actions/checkout@v2
+
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+
+      - name: Run quality checks
+        run: |
+          bin/quality
```
Comment (DmitryBarskov) [inline .github/workflows/ci.yml:40]: nbd
```suggestion
        run: bin/quality
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
+name: CI
+on:
+  pull_request:
+  push:
+    branches:
+      - 'master'
+      - 'github-actions'
```
Comment (DmitryBarskov) [inline .github/workflows/ci.yml:7]: Right now we have two identical workflows running. Maybe we just delete github-actions from here?

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+
+      - name: Run quality checks
```
Comment (DmitryBarskov) [inline .github/workflows/ci.yml:38]: I think we can run `bin/quality` separately (parallel) from tests. bin/quality doesn't need a pg. Maybe it is another job or even a workflow, not a step.

File: bin/quality [hunk 6]
```diff
@@ -5,4 +5,4 @@ set -e
 bin/rubocop
 bin/brakeman --quiet --skip-libs --exit-on-warn --no-pager
 bin/bundle-audit update
-bin/bundle-audit
+# bin/bundle-audit
```
Comment (DmitryBarskov) [inline bin/quality:8]: 🤔

File: .github/workflows/ci.yml (hunk 4 above)
Comment (EvgeniyEsaulkov) [reply .github/workflows/ci.yml:7]: I added `guthub-actions` branch for debugging purposes. Now it should be deleted. Thanks for noticing that 👍🏼

File: .github/workflows/ci.yml (hunk 5 above)
Comment (EvgeniyEsaulkov) [reply .github/workflows/ci.yml:38]: We don't need to run tests if quality check failed. That's why `bin/quality` and `bin/tests` are running in sequence

File: .github/workflows/ci.yml (hunk 1 above)
Comment (EvgeniyEsaulkov) [reply .github/workflows/ci.yml:44]: I'm happy for your projects but it's not related to this one. What is your proposal here?

Comment (DmitryBarskov) [reply .github/workflows/ci.yml:42]: 💩

Comment (DmitryBarskov) [reply .github/workflows/ci.yml:42]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    env:
+      DATABASE_URL: postgres://postgres:@localhost:5432/test
+      REDIS_URL: redis://localhost:6379/0
+      RAILS_ENV: test
+    services:
+      postgres:
+        image: postgres:11@sha256:85d79cba2d4942dad7c99f84ec389a5b9cc84fb07a3dcd3aff0fb06948cdc03b
```
Comment (DmitryBarskov) [inline .github/workflows/ci.yml:17]: Why do you use this very specific version?

File: .github/workflows/ci.yml (hunk 5 above)
Comment (RasimKhusaenov) [reply .github/workflows/ci.yml:37]: Why shouldn't we speed up CI if we have such an opportunity?

Comment (EvgeniyEsaulkov) [reply .github/workflows/ci.yml:37]: @RasimKhusaenov please read my comment above

Comment (ArthurZaharov) [reply .github/workflows/ci.yml:39]: @DmitryBarskov @RasimKhusaenov Created the issue https://github.com/fs/rails-base-graphql-api/issues/219 for this case. You can continue discussion there.

**Review by ArthurZaharov — APPROVED** [review]

**Pull request merged**